LedgerHQ/ledger-live-desktop
# added poc for progressive update mechanism

this PR feature a progressive update mechanism and a remote config system.

On app boot, the following data structure is fetched from github and loaded in memory: 

> https://github.com/LedgerHQ/ledger-live-desktop/blob/config/config.json

it then calculate a deterministic random local update threshold using the userId. If the threshold is inferior than the global one provided in the remote config, the update banner will appear. It is possible to define one global update threshold per platform (darwin, win32, linux ...).

Notes:

- the whole system is optional and if the config doesn't have the required data, the update will just appear normaly
- the deterministic random is based on a ${userId}-${version} seed, this way it's not always the same users that get the update first !

How to test ?

To test, you have to modify your package.json version number to something anterior to the last release (2.0.0 for instance), then you have to package the app (yarn dist).
Run the packaged app, and an update banner will appear (or not, 1 chance out of two with the actual config.json file)

Reset your userId in the localstorage if you want to reset the odds ;)
